src/modules/job-list/job_data.c

```diff
@@ -224,20 +249,39 @@ int job_parse_jobspec (struct job *job, const char *s)
     if (res[1].with && parse_res_level (job, res[1].with, &res[2]) < 0)
         goto nonfatal_error;
 
-    /* Set job->nnodes if available.  In jobspec version 1, only if
-     * resources listed as node->slot->core->NIL
-     */
     if (res[0].type != NULL && !strcmp (res[0].type, "node")
         && res[1].type != NULL && !strcmp (res[1].type, "slot")
         && res[2].type != NULL && !strcmp (res[2].type, "core")
-        && res[2].with == NULL)
+        && res[2].with == NULL) {
+        const char *type = NULL;
+        int count = 0;
+
+        /* Set job->nnodes b/c it is available.  In jobspec version 1,
+         * only if resources listed as node->slot->core->NIL
+         */
         job->nnodes = res[0].count;
 
-    /* Set job->ntasks
+        /* per-resource is used to overcome short-term gaps in
+         * Jobspec V1.  Remove per-resource logic below when it
+         * has been retired
+         */
+
+        if (parse_per_resource (job, &type, &count) < 0)
+            goto nonfatal_error;
+
+        /* if nodes specified, per-resource.type == "node", and
+         * per-resource.count > 0 there is an adjustment on ntasks.
+         */
+        if (type && streq (type, "node") && count > 0)
+            job->ntasks = res[0].count * count;
+    }
+
+    /* Set job->ntasks if not yet set
```

Comment:
Does per-resource type=core need to be handled here as well?

I wonder if this would be slightly clearer if `parse_per_resource()` returned success when it has a valid per-resource setting, and then the code for altering ntasks could be conditional on that rather than on the values of type and count?

Reply:
doh, this code is from the prior PR #4555 this was built on top of :-)  But it's a good suggestion.  I'll do that as a cleanup commit in this PR (need to rebase now that #4555 is merged).  And I can cleanup the thing you noticed below as well.

Reply:
> Does per-resource type=core need to be handled here as well?

Yeah, I think it does.  I was initially under the impression that `--tasks-per-core` would set ntasks the normal way, but that doesn't seem to be the case.  Good catch.